tests/Fixer/ControlStructure/YodaStyleFixerTest.php

```diff
@@ -0,0 +1,266 @@
+        );
+
+        $this->fixer->configure($config);
+        $this->doTest($expected, $input);
+
+        // test the reverse config
+        $config['yoda'] = false;
+        $this->fixer->configure($config);
+        if (null === $input) {
+            $this->doTest($expected);
+        } else {
+            $this->doTest($input, $expected);
+        }
+    }
+
+    public function provideFixCases()
+    {
+        return array(
+            // Don't fix cases.
+            array('<?php $a = 1 === 1;'),
+            array('<?php $a = $b === $c;'),
+            array('<?php $a = ((array(2))) == $a;'),
+            array('<?php $a = $a == ((array(2)));'),
+            array('<?php $a = array($a) != $a;'),
+            array('<?php $a = $a != array($a);'),
+            array('<?php $a = count($this->array[$var]) === $a;'),
+            array('<?php $a = $a === count($this->array[$var]);'),
+            array('<?php $a = ($a & self::MY_BITMASK) === $a;'),
+            array('<?php $a = $a === ($a & self::MY_BITMASK);'),
+            array('<?php $a = $this->getStuff() === $myVariable;'),
+            array('<?php $a = $myVariable === $this->getStuff();'),
+            array('<?php $a = 2 * $myVar % 3 === $a;'),
+            array('<?php return $a === 2 * $myVar % 3;'),
+            array('<?php $a = $c > 2;'),
+            // Fix cases.
+            array(
+                '<?php $z = 2 == $a; $b = $c > 1 && $c <= 10;',
+                '<?php $z = $a == 2; $b = $c > 1 && $c <= 10;',
+            ),
+            array(
+                '<?php $z = /**/2
+                 # aa
+                 == $a/***/;',
+                '<?php $z = /**/$a
+                 # aa
+                 == 2/***/;',
+            ),
+            array(
+                '<?php return 2 == ($a)?>',
+                '<?php return ($a) == 2?>',
+            ),
+            array(
+                '<?php $a = ($c === ((null === $b)));',
+                '<?php $a = ($c === (($b === null)));',
+            ),
+            array(
+                '<?php return null == $a[2];',
+                '<?php return $a[2] == null;',
+            ),
+            array(
+                '<?php return "" === $this->myArray[$index];',
+                '<?php return $this->myArray[$index] === "";',
+            ),
+            array(
+                '<?php return "" === $this->myArray[$index]->a;',
+                '<?php return $this->myArray[$index]->a === "";',
+            ),
+            array(
+                '<?php return "" === $this->myObject->{$index};',
+                '<?php return $this->myObject->{$index} === "";',
+            ),
+            array(
+                '<?php return "" === $this->myObject->{$index}->a;',
+                '<?php return $this->myObject->{$index}->a === "";',
+            ),
+            array(
+                '<?php return self::MY_CONST === self::$myVariable;',
+                '<?php return self::$myVariable === self::MY_CONST;',
```

Comment:
The issue is that Yoda is supposed to shield you from doing an assignment when you don't need to. But here, since you will get an error message as show in my previous comment, it is not worth it, which means it might be better to use "natural" (at least for english people) order. We're not from Dagobah after all. So to sum up, Yoda should only apply IMO to expressions like `something == 42` (or even `$x === 42`, why not), if and only if `something = 42` does not throw an error message.